compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2513,6 +2539,54 @@ static LogicalResult setRootConfig(mlir::FunctionOpInterface entryPointFn,
       entryPointFn, op, tileSizes, DispatchLoweringPassPipeline::CPUDefault);
 }
 
+// Returns true if TopkOp can be lowered with SCF dialect,
+// otherwise false.
+static bool canTopkBeLoweredUsingSCF(mlir::FunctionOpInterface entryPointFn,
+                                     IREE::LinalgExt::TopkOp topkOp) {
+  if (!clSCFLowerTopk)
+    return false;
+  auto arg0 = topkOp.getInputs()[0];
+  auto shapedTy = llvm::cast<ShapedType>(arg0.getType());
+  SmallVector<int64_t> vectorSizes;
+  vectorSizes.append(shapedTy.getShape().begin(), shapedTy.getShape().end());
```

Comment:
nit1: use `cast<>` directly for consistency.

nit2: You can collapse `append` with initializer, e.g.,

```suggestion
  auto shapedTy = cast<ShapedType>(arg0.getType());
  SmallVector<int64_t> vectorSizes(shapedTy.getShape());
```